src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -432,8 +432,11 @@ void QgsMeshLayer::setRendererSettings( const QgsMeshRendererSettings &settings
   if ( oldActiveVector != mRendererSettings.activeVectorDatasetGroup() )
     emit activeVectorDatasetGroupChanged( mRendererSettings.activeVectorDatasetGroup() );
 
-  emit rendererChanged();
-  triggerRepaint();
+  if ( repaint )
+  {
+    emit rendererChanged();
```

Comment:
Do we need to skip emitting the signal too?
If we keep it we can skip the `emit meshLayer->legendChanged();` from qgisapp

Reply:
Ok, that changed those. This signal is raised and the `legendChanged` does not